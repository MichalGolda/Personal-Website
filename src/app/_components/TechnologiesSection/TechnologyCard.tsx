export type TechnologyCardProps = {
  title: string;
  content: () => React.ReactElement;
  icons?: React.ReactElement;
};

const TechnologyCard: React.FC<TechnologyCardProps> = ({
  title,
  content,
  icons,
}) => {
  return (
    <div className="w-full flex flex-col gap-y-4">
      <h4 className="text-lg md:text-xl lg:text-2xl text-secondary font-bold">
        {title}
      </h4>
      <p className="text-sm md:text-base text-body">{content()}</p>
      {icons && <div className="flex flex-row flex-wrap gap-4">{icons}</div>}
    </div>
  );
};

export default TechnologyCard;
