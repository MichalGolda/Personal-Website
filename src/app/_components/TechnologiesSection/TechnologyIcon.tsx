import Image from "next/image";

export type TechnologyIconProps = {
  src: string;
  alt: string;
};

const TechnologyIcon = ({ src, alt }: TechnologyIconProps) => {
  return (
    <Image
      key={src}
      className="max-w-8 max-h-8"
      src={src}
      alt={alt}
      width={32}
      height={32}
    />
  );
};

export const TypescriptIcon = () => (
  <TechnologyIcon src="/typescript.webp" alt="typescript" />
);

export const JavascriptIcon = () => (
  <TechnologyIcon src="/javascript.webp" alt="javascript" />
);

export const ReactIcon = () => <TechnologyIcon src="/react.webp" alt="react" />;

export const VueIcon = () => <TechnologyIcon src="/vue.webp" alt="vue" />;

export const NextJsIcon = () => (
  <TechnologyIcon src="/nextjs.webp" alt="nextjs" />
);

export const NuxtJsIcon = () => (
  <TechnologyIcon src="/nuxtjs.webp" alt="nuxtjs" />
);

export const SassIcon = () => <TechnologyIcon src="/sass.webp" alt="sass" />;

export const TailwindCssIcon = () => (
  <TechnologyIcon src="/tailwindcss.webp" alt="tailwindcss" />
);

export const DjangoIcon = () => (
  <TechnologyIcon src="/django.webp" alt="django" />
);

export const PHPIcon = () => <TechnologyIcon src="/php.webp" alt="php" />;

export const DockerIcon = () => (
  <TechnologyIcon src="/docker.webp" alt="docker" />
);

export const NodeJsIcon = () => (
  <TechnologyIcon src="/nodejs.webp" alt="nodejs" />
);

export const NestJsIcon = () => (
  <TechnologyIcon src="/nestjs.webp" alt="nestjs" />
);

export const PostgreSQLIcon = () => (
  <TechnologyIcon src="/postgresql.webp" alt="postgresql" />
);

export const AWSIcon = () => <TechnologyIcon src="/aws.webp" alt="aws" />;

export const GCPIcon = () => <TechnologyIcon src="/gcp.webp" alt="gcp" />;

export const AzureIcon = () => <TechnologyIcon src="/azure.webp" alt="azure" />;

export const PrettierIcon = () => (
  <TechnologyIcon src="/prettier.webp" alt="prettier" />
);

export const EslintIcon = () => (
  <TechnologyIcon src="/eslint.webp" alt="eslint" />
);

export const FigmaIcon = () => <TechnologyIcon src="/figma.webp" alt="figma" />;

export default TechnologyIcon;
