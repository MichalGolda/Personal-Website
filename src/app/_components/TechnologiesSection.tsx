import { SectionHeading } from "./SectionHeading";
import TechnologyCard from "./TechnologyCard";

export default function TechnologiesSection() {
  return (
    <section className="max-w-[1320px] mx-auto my-64">
      <SectionHeading
        headline="technologie"
        title="Pracuje z technologiami przyszłości"
        body="Stosuje najnowsze technologie i rozwiązania używane w zaawansowanych aplikacjach webowych."
      />
      <div className="grid grid-rows-2 grid-cols-2 gap-16 mt-16">
        <TechnologyCard
          title="Frontend"
          content="Spójny, wykonany z dbałością o szczegóły, SEO-friendy, responsywny, działający na dowolnym urządzeniu: TypeScript, React, Next.js, Nuxt.js, Headless."
        />
        <TechnologyCard
          title="Backend"
          content="Wydajność, skalowalność, globalna dostępność i bezpieczeństwo. Osiągamy to dzięki sprawdzonym rozwiązaniom i frameworkom: Python, Django, PHP, Laravel, Wordpress, PosgreSQL, DynamoDB, AWS, Docker, integrujemy się z dowolnym systemem za pomocą API, budujemy aplikacje oparte o mikrousługi."
        />
        <TechnologyCard
          title="QA"
          content="Jakość kodu i finalnego produktu zapewniam dzięki Clean code, Unit tests, Selenium, Code Review."
        />
        <TechnologyCard
          title="Design UX/UI"
          content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ut purus congue, varius neque posuere, egestas magna. "
        />
      </div>
    </section>
  );
}
