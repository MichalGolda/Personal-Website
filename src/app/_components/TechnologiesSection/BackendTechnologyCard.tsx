import TechnologyCard from "./TechnologyCard";
import {
  AWSIcon,
  AzureIcon,
  DjangoIcon,
  DockerIcon,
  GCPIcon,
  JavascriptIcon,
  NestJsIcon,
  NodeJsIcon,
  PostgreSQLIcon,
  TypescriptIcon,
} from "./TechnologyIcon";

const Icons = () => {
  return (
    <>
      <TypescriptIcon />
      <JavascriptIcon />
      <DjangoIcon />
      <DockerIcon />
      <NodeJsIcon />
      <NestJsIcon />
      <PostgreSQLIcon />
      <AWSIcon />
      <GCPIcon />
      <AzureIcon />
    </>
  );
};

const BackendTechnologyCard = () => {
  return (
    <TechnologyCard
      title="Backend"
      content={() => (
        <>
          Dostępność, skalowalność osiągam dzięki sprawdzonym rozwiązaniom:{" "}
          <b className="text-primary">Python</b>,{" "}
          <b className="text-primary">Django</b>,{" "}
          <b className="text-primary">PHP</b>,{" "}
          <b className="text-primary">Wordpress</b>,{" "}
          <b className="text-primary">Docker</b>,{" "}
          <b className="text-primary">NodeJs</b>,{" "}
          <b className="text-primary">Nest.js</b>,{" "}
          <b className="text-primary">PostgreSQL</b>,{" "}
          <b className="text-primary">AWS</b>,{" "}
          <b className="text-primary">GCP</b>,{" "}
          <b className="text-primary">Azure</b>.
        </>
      )}
      icons={<Icons />}
    />
  );
};

export default BackendTechnologyCard;
