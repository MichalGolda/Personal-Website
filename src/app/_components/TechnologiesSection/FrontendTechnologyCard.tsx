import TechnologyCard from "./TechnologyCard";
import {
  JavascriptIcon,
  NextJsIcon,
  NuxtJsIcon,
  ReactIcon,
  SassIcon,
  TailwindCssIcon,
  TypescriptIcon,
  VueIcon,
} from "./TechnologyIcon";

const Icons = () => {
  return (
    <>
      <TypescriptIcon />
      <JavascriptIcon />
      <ReactIcon />
      <VueIcon />
      <NextJsIcon />
      <NuxtJsIcon />
      <SassIcon />
      <TailwindCssIcon />
    </>
  );
};

const FrontendTechnologyCard = () => {
  return (
    <TechnologyCard
      title="Frontend"
      content={() => (
        <>
          Wykonany z dbałością o szczegóły, seo-friendly, spójny, działający na
          dowolnym urządzeniu: <b className="text-primary">Javascript</b>,{" "}
          <b className="text-primary">Typescript</b>,{" "}
          <b className="text-primary">React</b>,{" "}
          <b className="text-primary">Vue</b>,{" "}
          <b className="text-primary">Next.js</b>,{" "}
          <b className="text-primary">Nuxt.js</b>,{" "}
          <b className="text-primary">sass</b>,{" "}
          <b className="text-primary">tailwindcss</b>.
        </>
      )}
      icons={<Icons />}
    />
  );
};

export default FrontendTechnologyCard;
