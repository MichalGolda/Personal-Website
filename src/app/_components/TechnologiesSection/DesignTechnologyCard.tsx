import TechnologyCard from "./TechnologyCard";
import { FigmaIcon } from "./TechnologyIcon";

const DesignTechnologyCard = () => {
  return (
    <TechnologyCard
      title="Design UX/UI"
      content={
        <>
          Spójny i z dbałością o szczegóły projekt przygotowuję z pomocą:{" "}
          <b className="text-primary">Figma</b>
        </>
      }
      icons={<FigmaIcon />}
    />
  );
};

export default DesignTechnologyCard;
