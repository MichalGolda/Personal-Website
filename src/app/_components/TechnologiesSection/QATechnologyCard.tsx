import TechnologyCard from "./TechnologyCard";
import { EslintIcon, PrettierIcon } from "./TechnologyIcon";

const Icons = () => {
  return (
    <>
      <PrettierIcon />
      <EslintIcon />
    </>
  );
};

const QATechnologyCard = () => {
  return (
    <TechnologyCard
      title="QA"
      content={() => (
        <>
          Jakość kodu i finalnego projektu zapewniam dzięki:{" "}
          <b className="text-primary">Clean Code</b>,{" "}
          <b className="text-primary">KISS</b>,{" "}
          <b className="text-primary">DRY</b>,{" "}
          <b className="text-primary">YAGNI</b>,{" "}
          <b className="text-primary">Testy jednostkowe</b>,{" "}
          <b className="text-primary">Testy integracyjne</b>,{" "}
          <b className="text-primary">prettier</b>,{" "}
          <b className="text-primary">eslint</b>.
        </>
      )}
      icons={<Icons />}
    />
  );
};

export default QATechnologyCard;
