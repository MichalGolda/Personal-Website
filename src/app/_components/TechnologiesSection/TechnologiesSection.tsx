import { Container, SectionHeading } from "@/app/_components";
import TechnologyCard from "./TechnologyCard";

export default function TechnologiesSection() {
  return (
    <section id="technologies" className="my-64">
      <Container>
        <SectionHeading
          headline="technologie"
          title="Pracuje z technologiami przyszłości"
          body="Stosuje najnowsze technologie i rozwiązania używane w zaawansowanych aplikacjach webowych."
        />
        <div className="grid grid-rows-2 grid-cols-1 md:grid-cols-2 gap-16 mt-16">
          <TechnologyCard
            title="Frontend"
            content="Spójny, wykonany z dbałością o szczegóły, SEO-friendy, responsywny, działający na dowolnym urządzeniu: TypeScript, React, Next.js, Nuxt.js, Headless."
          />
          <TechnologyCard
            title="Backend"
            content="Wydajno, skalowalno, globalna dostpno i bezpieczestwo. Osigamy to dziki sprawdzonym rozwizaniom i frameworkom:Python,Django,PHP,Laravel,
            Wordpress,PosgreSQL,
            DynamoDB,AWS, Docker, integrujemy si z dowolnym systemem za pomoc API, budujemy aplikacje oparte o mikrousugi."
          />
          <TechnologyCard
            title="QA"
            content="Jakość kodu i finalnego produktu zapewniam dzięki Clean code, Unit tests, Selenium, Code Review."
          />
          <TechnologyCard
            title="Design UX/UI"
            content="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum ut purus congue, varius neque posuere, egestas magna. "
          />
        </div>
      </Container>
    </section>
  );
}
