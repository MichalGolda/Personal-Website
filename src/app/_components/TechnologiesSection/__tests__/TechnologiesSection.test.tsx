import "@/app/_mocks/intersectionObserver.mock";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import TechnologiesSection from "../TechnologiesSection";

describe("TechnologiesSection component", () => {
  it("renders unchanged", () => {
    const { container } = render(<TechnologiesSection />);

    expect(container).toMatchSnapshot();
  });
});
