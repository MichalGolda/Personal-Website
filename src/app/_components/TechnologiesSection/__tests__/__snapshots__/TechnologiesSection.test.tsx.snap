// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologiesSection component renders unchanged 1`] = `
<div>
  <section
    class="my-64"
    id="technologies"
  >
    <div
      class="md:max-w-[1024px] lg:max-w-[1320px] w-full mx-auto md:px-16 px-4 "
    >
      <div
        class="text-center max-w-3xl mx-auto"
      >
        <span
          class="text-primary font-semibold text-md md:text-lg lg:text-xl uppercase"
        >
          technologie
        </span>
        <h2
          class="text-2xl md:text-3xl lg:text-4xl text-secondary font-bold"
        >
          Pracuje z technologiami przyszłości
        </h2>
        <p
          class="text-sm md:text-base text-body mt-4"
        >
          Stosuje najnowsze technologie i rozwiązania używane w zaawansowanych aplikacjach webowych.
        </p>
      </div>
      <div
        class="grid grid-rows-2 grid-cols-1 md:grid-cols-2 gap-16 mt-16"
      >
        <div
          class="w-full flex flex-col gap-y-4"
        >
          <h4
            class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
          >
            Frontend
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Wykonany z dbałością o szczegóły, seo-friendly, spójny, działający na dowolnym urządzeniu: 
            <b
              class="text-primary"
            >
              Javascript
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Typescript
            </b>
            ,
             
            <b
              class="text-primary"
            >
              React
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Vue
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Next.js
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Nuxt.js
            </b>
            ,
             
            <b
              class="text-primary"
            >
              sass
            </b>
            ,
             
            <b
              class="text-primary"
            >
              tailwindcss
            </b>
            .
          </p>
          <div
            class="flex flex-row flex-wrap gap-4"
          >
            <img
              alt="typescript"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Ftypescript.webp&w=64&q=75"
              srcset="/_next/image?url=%2Ftypescript.webp&w=32&q=75 1x, /_next/image?url=%2Ftypescript.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="javascript"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fjavascript.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fjavascript.webp&w=32&q=75 1x, /_next/image?url=%2Fjavascript.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="react"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Freact.webp&w=64&q=75"
              srcset="/_next/image?url=%2Freact.webp&w=32&q=75 1x, /_next/image?url=%2Freact.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="vue"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fvue.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fvue.webp&w=32&q=75 1x, /_next/image?url=%2Fvue.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="nextjs"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fnextjs.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fnextjs.webp&w=32&q=75 1x, /_next/image?url=%2Fnextjs.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="nuxtjs"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fnuxtjs.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fnuxtjs.webp&w=32&q=75 1x, /_next/image?url=%2Fnuxtjs.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="sass"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fsass.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fsass.webp&w=32&q=75 1x, /_next/image?url=%2Fsass.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="tailwindcss"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Ftailwindcss.webp&w=64&q=75"
              srcset="/_next/image?url=%2Ftailwindcss.webp&w=32&q=75 1x, /_next/image?url=%2Ftailwindcss.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
        <div
          class="w-full flex flex-col gap-y-4"
        >
          <h4
            class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
          >
            Backend
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Dostępność, skalowalność osiągam dzięki sprawdzonym rozwiązaniom:
             
            <b
              class="text-primary"
            >
              Python
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Django
            </b>
            ,
             
            <b
              class="text-primary"
            >
              PHP
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Wordpress
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Docker
            </b>
            ,
             
            <b
              class="text-primary"
            >
              NodeJs
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Nest.js
            </b>
            ,
             
            <b
              class="text-primary"
            >
              PostgreSQL
            </b>
            ,
             
            <b
              class="text-primary"
            >
              AWS
            </b>
            ,
             
            <b
              class="text-primary"
            >
              GCP
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Azure
            </b>
            .
          </p>
          <div
            class="flex flex-row flex-wrap gap-4"
          >
            <img
              alt="typescript"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Ftypescript.webp&w=64&q=75"
              srcset="/_next/image?url=%2Ftypescript.webp&w=32&q=75 1x, /_next/image?url=%2Ftypescript.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="javascript"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fjavascript.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fjavascript.webp&w=32&q=75 1x, /_next/image?url=%2Fjavascript.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="django"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fdjango.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fdjango.webp&w=32&q=75 1x, /_next/image?url=%2Fdjango.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="docker"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fdocker.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fdocker.webp&w=32&q=75 1x, /_next/image?url=%2Fdocker.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="nodejs"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fnodejs.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fnodejs.webp&w=32&q=75 1x, /_next/image?url=%2Fnodejs.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="nestjs"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fnestjs.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fnestjs.webp&w=32&q=75 1x, /_next/image?url=%2Fnestjs.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="postgresql"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fpostgresql.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fpostgresql.webp&w=32&q=75 1x, /_next/image?url=%2Fpostgresql.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="aws"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Faws.webp&w=64&q=75"
              srcset="/_next/image?url=%2Faws.webp&w=32&q=75 1x, /_next/image?url=%2Faws.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="gcp"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fgcp.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fgcp.webp&w=32&q=75 1x, /_next/image?url=%2Fgcp.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="azure"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fazure.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fazure.webp&w=32&q=75 1x, /_next/image?url=%2Fazure.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
        <div
          class="w-full flex flex-col gap-y-4"
        >
          <h4
            class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
          >
            QA
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Jakość kodu i finalnego projektu zapewniam dzięki:
             
            <b
              class="text-primary"
            >
              Clean Code
            </b>
            ,
             
            <b
              class="text-primary"
            >
              KISS
            </b>
            ,
             
            <b
              class="text-primary"
            >
              DRY
            </b>
            ,
             
            <b
              class="text-primary"
            >
              YAGNI
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Testy jednostkowe
            </b>
            ,
             
            <b
              class="text-primary"
            >
              Testy integracyjne
            </b>
            ,
             
            <b
              class="text-primary"
            >
              prettier
            </b>
            ,
             
            <b
              class="text-primary"
            >
              eslint
            </b>
            .
          </p>
          <div
            class="flex flex-row flex-wrap gap-4"
          >
            <img
              alt="prettier"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fprettier.webp&w=64&q=75"
              srcset="/_next/image?url=%2Fprettier.webp&w=32&q=75 1x, /_next/image?url=%2Fprettier.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <img
              alt="eslint"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Feslint.webp&w=64&q=75"
              srcset="/_next/image?url=%2Feslint.webp&w=32&q=75 1x, /_next/image?url=%2Feslint.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
        <div
          class="w-full flex flex-col gap-y-4"
        >
          <h4
            class="text-lg md:text-xl lg:text-2xl text-secondary font-bold"
          >
            Design UX/UI
          </h4>
          <p
            class="text-sm md:text-base text-body"
          >
            Spójny i z dbałością o szczegóły projekt przygotowuję z pomocą:
             
            <b
              class="text-primary"
            >
              Figma
            </b>
          </p>
          <div
            class="flex flex-row flex-wrap gap-4"
          >
            <img
              alt="figma"
              class="max-w-8 max-h-8"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Ffigma.webp&w=64&q=75"
              srcset="/_next/image?url=%2Ffigma.webp&w=32&q=75 1x, /_next/image?url=%2Ffigma.webp&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
